### PR TITLE
952 refactor endpoints

### PR DESCRIPTION
**Pull request by BCerki**
card: https://github.com/orgs/bcgov/projects/122/views/2?filterQuery=&pane=issue&itemId=54057669

Notes:
- We have a separate card about renaming endpoints according to best practices: https://github.com/bcgov/cas-registration/issues/1077 (the refactor revealed some inconsistent naming)
- We don't have a `user_operator` endpoint--everything is more specific--so there wasn't an example of what to do with operations/operators etc. I made a file with the plural, and a folder for the singular for anything that's more specific. Not sure about this approach--thoughts very welcome!

### CODE REVIEW

File: bc_obps/registration/api/operation/update_status.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+from uuid import UUID
+from registration.constants import PAGE_SIZE, UNAUTHORIZED_MESSAGE
+from django.db import transaction
+from registration.decorators import authorize
+from registration.api.api_base import router
+from datetime import datetime
+from django.core.exceptions import ValidationError
+import pytz
```
[the remark this answers is not in the document]
Comment (BCerki) [reply bc_obps/registration/api/operation/update_status.py:8]: we have a tech debt card for this

File: bc_obps/registration/api/users.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+from registration.api.api_base import router
+
+##### GET #####
+
+
+@router.get("/user", response=UserOut, url_name="get_user_by_guid")
+@authorize(AppRole.get_all_authorized_app_roles(), UserOperator.get_all_industry_user_operator_roles(), False)
+def get_user_by_guid(request):
+    user: User = request.current_user
+    return user
```
[the remark this answers is not in the document]
Comment (BCerki) [reply bc_obps/registration/api/users.py:13]: Is this a problem? I just renamed the file; I didn't change any code

**Comment by Sepehr-Sobhani** [comment]
> * We don't have a `user_operator` endpoint--everything is more specific--so there wasn't an example of what to do with operations/operators etc. I made a file with the plural, and a folder for the singular for anything that's more specific. Not sure about this approach--thoughts very welcome!

I suggest having API folder structure that reflects the structure of the endpoint, for example for `Operations` related endpoints we could have endpoints like:
- get all operations => `router.get("/operations")`
- get single operation => `router.get("/operations/{operation_id}")`
-  update single operation => `router.put("/operations/{operation_id}")`
- create operation => `router.post("/operations")`
- update status of operation => `router.put("/operations/{operation_id}/status")`
So, all these endpoints start with `operations` so I think it makes more sense to put them all in a folder called `operations` and split them into separate files based on each endpoint. (The confusing part here would be how to name each file 🤔)

@pbastia feel free to correct me if I'm wrong.

**Review by Sepehr-Sobhani — APPROVED** [review]
Looks great! 🤩 Just added some nits (as always 😃 ) and shared my thoughts on folder structures otherwise looks great.

File: bc_obps/registration/api/operation/update_status.py
[the remark this answers is not in the document]
Comment (BCerki) [reply bc_obps/registration/api/operation/update_status.py:1]: Are you using a tool to show you these? In the frontend I have a key command to remove unused imports but it's not working for python files

[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply bc_obps/registration/api/operation/update_status.py:1]: No, I only notice that when I open the file in my editor :) This should be captured by Pylint once we set it up properly.

**Pull request merged**